src/set_theory/surreal/dyadic.lean

```diff
@@ -30,129 +31,122 @@ local infix ` ≈ ` := pgame.equiv
+(pow_half_succ_lt_pow_half n).trans_le $ pow_half_le_one n
 
-theorem zero_le_pow_half {n : ℕ} : 0 ≤ pow_half n :=
-zero_lt_pow_half.le
+theorem pow_half_pos (n : ℕ) : 0 < pow_half n :=
+by { rw [←lf_iff_lt numeric_zero (numeric_pow_half n), zero_lf_le], simp }
 
-theorem add_pow_half_succ_self_eq_pow_half {n} : pow_half (n + 1) + pow_half (n + 1) ≈ pow_half n :=
+theorem zero_le_pow_half (n : ℕ) : 0 ≤ pow_half n :=
+(pow_half_pos n).le
+
+theorem add_pow_half_succ_self_eq_pow_half (n) : pow_half (n + 1) + pow_half (n + 1) ≈ pow_half n :=
 begin
-  induction n with n hn,
-  { exact half_add_half_equiv_one },
+  induction n using nat.strong_induction_on with n hn,
   { split; rw le_iff_forall_lf; split,
     { rintro (⟨⟨ ⟩⟩ | ⟨⟨ ⟩⟩); apply lf_of_lt,
-      { calc 0 + pow_half (n.succ + 1) ≈ pow_half (n.succ + 1) : zero_add_equiv _
-                                   ... < pow_half n.succ       : pow_half_succ_lt_pow_half },
-      { calc pow_half (n.succ + 1) + 0 ≈ pow_half (n.succ + 1) : add_zero_equiv _
-                                   ... < pow_half n.succ       : pow_half_succ_lt_pow_half } },
-    { rintro ⟨ ⟩,
-      rw lf_iff_forall_le,
-      right,
-      use sum.inl punit.star,
-      calc pow_half (n.succ) + pow_half (n.succ + 1)
-          ≤ pow_half (n.succ) + pow_half (n.succ) : add_le_add_left pow_half_succ_le_pow_half _
-      ... ≈ pow_half n                            : hn },
-    { rintro ⟨ ⟩, apply lf_of_lt,
-      calc 0 ≈ 0 + 0                                        : (add_zero_equiv _).symm
-        ... ≤ pow_half (n.succ + 1) + 0                     : add_le_add_right zero_le_pow_half _
-        ... < pow_half (n.succ + 1) + pow_half (n.succ + 1) : add_lt_add_left zero_lt_pow_half _ },
+      { apply lt_of_equiv_of_lt (zero_add_equiv _) (pow_half_succ_lt_pow_half n) },
+      { apply lt_of_equiv_of_lt (add_zero_equiv _) (pow_half_succ_lt_pow_half n) } },
+    { cases n, { rintro ⟨ ⟩ }, 
+      rintro ⟨ ⟩,
+      refine lf_of_forall_le (or.inr ⟨sum.inl punit.star, _⟩),
+      apply le_of_le_of_equiv (add_le_add_left (pow_half_succ_le_pow_half _) _)
+        (hn _ (nat.lt_succ_self n)) },
+    { simp only [pow_half_move_left, forall_const],
+      apply lf_of_lt,
+      calc 0 ≈ 0 + 0                            : (add_zero_equiv 0).symm
+        ... ≤ pow_half n.succ + 0               : add_le_add_right (zero_le_pow_half _) _
+        ... < pow_half n.succ + pow_half n.succ : add_lt_add_left (pow_half_pos _) _ },
     { rintro (⟨⟨ ⟩⟩ | ⟨⟨ ⟩⟩); apply lf_of_lt,
-      { calc pow_half n.succ
-            ≈ pow_half n.succ + 0                        : (add_zero_equiv _).symm
-        ... < pow_half (n.succ) + pow_half (n.succ + 1)  : add_lt_add_left zero_lt_pow_half _ },
-      { calc pow_half n.succ
-            ≈ 0 + pow_half n.succ                        : (zero_add_equiv _).symm
-        ... < pow_half (n.succ + 1) + pow_half (n.succ)  : add_lt_add_right zero_lt_pow_half _ } } }
```

Comment:
It feels like a shame to lose these calc proofs

Reply:
I don't think there's a need to use a `calc` block when we can just call the single transitivity theorem explicitly.

Reply:
Sure there is, it makes the proof easier for a human to read.

Reply:
I think that the `calc` blocks I removed are all simple enough that this shouldn't be an issue.

Reply:
I've asked the other maintainers for a tie-break (which means I won't be merging this tonight)

Reply:
I think all the type hints from a `calc` block tend to make things easier to maintain. There's no harm having this small amount of redundancy here.

I think golfing is best limited to when it's clear from the code what the types are supposed to be, for instance golfing the entire proof of a lemma, or golfing the proof of a `have` block. If you're feeling like you need to golf some goal in the middle of a longer proof then that suggests to me that there are some missing lemmas.

Reply:
Alright then, I'll revert shortly.